src/proto/pi/mp.rs

```diff
@@ -0,0 +1,199 @@
+//! Multi-processor management protocols.
+
+use crate::proto::Protocol;
+use crate::{unsafe_guid, Result, Status};
+use bitflags::bitflags;
+use core::convert::TryInto;
+use core::ffi::c_void;
+use core::ptr;
+use core::time::Duration;
+
+/// Callback to be called on the AP.
+pub type Procedure = extern "win64" fn(*mut c_void);
+
+bitflags! {
+    /// Flags indicating if the processor is BSP or AP,
+    /// if the processor is enabled or disabled, and if
+    /// the processor is healthy.
+    #[derive(Default)]
+    pub struct StatusFlag: u32 {
```

Comment:
Instead of exposing these flags, could we get them as methods such as `is_bsp`, `is_enabled` etc. on `ProcessorInformation`?

Reply:
See a347021.

Also, perhaps we should wrap `CPUPhysicalLocation` and the two remaining struct fields too?

Reply:
No, I think this is ergonomic enough